chia/util/config.py

```diff
@@ -43,13 +48,14 @@ def config_path_for_filename(root_path: Path, filename: Union[str, Path]) -> Pat
 
 def save_config(root_path: Path, filename: Union[str, Path], config_data: Any):
     path: Path = config_path_for_filename(root_path, filename)
-    tmp_path: Path = path.with_suffix("." + str(os.getpid()))
-    with open(tmp_path, "w") as f:
-        yaml.safe_dump(config_data, f)
-    try:
-        os.replace(str(tmp_path), path)
-    except PermissionError:
-        shutil.move(str(tmp_path), str(path))
+    with tempfile.TemporaryDirectory(dir=path.parent) as tmp_dir:
+        tmp_path: Path = Path(tmp_dir) / (Path(filename).with_suffix("." + str(os.getpid())))
+        with open(tmp_path, "w") as f:
+            yaml.safe_dump(config_data, f)
+        try:
+            os.replace(str(tmp_path), path)
+        except PermissionError:
```

Comment:
Interestingly, the docs for `tempFile.TemporaryDirectory` mention PermissionError being raised on Windows if the file is already open:

"The directory can be explicitly cleaned up by calling the cleanup() method. If ignore_cleanup_errors is true, any unhandled exceptions during explicit or implicit cleanup (such as a [PermissionError](https://docs.python.org/3/library/exceptions.html#PermissionError) removing open files on Windows) will be ignored, and the remaining removable items deleted on a “best-effort” basis. Otherwise, errors will be raised in whatever context cleanup occurs (the cleanup() call, exiting the context manager, when the object is garbage-collected or during interpreter shutdown)."

I wonder if PermissionError is being seen more commonly now that the config is being (opened and) read more frequently. If this change doesn't completely fix the issue, maybe we should consider the same type of retry loop that you've implemented in `load_config`.